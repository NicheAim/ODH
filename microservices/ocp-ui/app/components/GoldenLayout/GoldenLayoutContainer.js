import styled from 'styled-components';
import PropTypes from 'prop-types';

const GoldenLayoutContainer = styled.div`
  /* Main golden layout container */
  & .lm_goldenlayout {
    background: transparent !important;
  }

  /* Set height considering app bar height */
  & .lm_controls {
    margin-top: 10px;
  }

  & .lm_header + .lm_items {
    background: ${(props) => props.secondaryColor};
    border: 1px solid #3275c1;
    border-top: 0;
    border-bottom-left-radius: 5px;
    border-bottom-right-radius: 5px;
  }

  & .lm_header {
    height: 30px !important;
  }

  & .lm_header + .lm_items,
  .lm_item_container {
    width: auto !important;
    height: ${(props) => props.contentHeight};
  }

  & .lm_root {
    height: ${(props) => props.containerHeight};
    width: ${(props) => props.containerWidth};
  }

  /* Enable scroll content */
  & .lm_content {
    overflow: auto;
    max-height: 98%;
    border-top: 1px solid #3275c1;
    border-bottom: 0 !important;
    border-left: 0 !important;
    border-right: 0 !important;
    background: transparent;
  }

  & .lm_tab {
    height: 18px !important;
    margin-top: 5px !important;
    border-radius: 5px;
    border: 1px solid #3275c1 !important;
    border-bottom-left-radius: 0;
    border-bottom-right-radius: 0;
    box-shadow: none !important;
  }

  & .lm_tab.lm_active {
    color: ${(props) => props.secondaryColor};
    background: #3275c1;
    font-size: 13px;
    font-family: Arial Bold, Arial, sans-serif;
  }

  & .lm_tab.lm_active:hover,
  .lm_tab:hover {
    color: ${(props) => props.secondaryColor};
    font-size: 13px;
    background: #099;
  }

  & .lm_title {
    text-transform: uppercase;
  }

  & .ui-hide { /* class used to move things off the screen but not hide them, this is for 508 compliance */
    position: absolute !important;
    left: -999999px !important;
    height: 0 !important;
    top: 0;
    padding: 0 !important;
    margin: 0 !important;
  }
`;

GoldenLayoutContainer.propTypes = {
  containerHeight: PropTypes.string,
  containerWidth: PropTypes.string,
  contentHeight: PropTypes.string,
  primaryColor: PropTypes.string,
  secondaryColor: PropTypes.string,
};

GoldenLayoutContainer.defaultProps = {
  containerHeight: 'calc(100vh - 75px)',
  containerWidth: '100vw',
  primaryColor: '#33666f',
  secondaryColor: '#ffffff',
  contentHeight: 'auto',
};

export default GoldenLayoutContainer;
