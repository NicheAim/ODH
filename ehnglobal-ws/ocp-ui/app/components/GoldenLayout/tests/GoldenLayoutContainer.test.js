/*
import React from 'react';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-15';
import 'jest-styled-components';

import GoldenLayoutContainer from '../GoldenLayoutContainer';

configure({ adapter: new Adapter() });

describe('<GoldenLayoutContainer />', () => {
  describe('snapshot tests', () => {
    it('should match snapshot', () => {
      // Act
      const renderedComponent = shallow(<GoldenLayoutContainer />);

      // Assert
      expect(renderedComponent).toMatchSnapshot();
    });
  });

  describe('styling tests', () => {
    it('should have GoldenLayout root style to reduce the height by considering app bar height', () => {
      // Act
      const renderedComponent = shallow(<GoldenLayoutContainer />);

      // Assert
      expect(renderedComponent).toHaveStyleRule('height', 'calc(100vh - 75px)', {
        modifier: ' .lm_root',
      });
    });

    it('should have GoldenLayout content style to add scrollbar', () => {
      // Act
      const renderedComponent = shallow(<GoldenLayoutContainer />);

      // Assert
      expect(renderedComponent).toHaveStyleRule('overflow', 'auto', {
        modifier: ' .lm_content',
      });
    });

    it('should have GoldenLayout title style to set to uppercase', () => {
      // Act
      const renderedComponent = shallow(<GoldenLayoutContainer />);

      // Assert
      expect(renderedComponent).toHaveStyleRule('text-transform', 'uppercase', {
        modifier: ' .lm_title',
      });
    });
  });
});
*/
