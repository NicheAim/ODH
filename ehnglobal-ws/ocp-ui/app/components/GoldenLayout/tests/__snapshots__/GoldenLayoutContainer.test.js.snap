// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<GoldenLayoutContainer /> snapshot tests should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GoldenLayoutContainer__GoldenLayoutContainer
    containerHeight="calc(100vh - 75px)"
    containerWidth="100vw"
    contentHeight="auto"
    primaryColor="#33666f"
    secondaryColor="#ffffff"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "className": "GoldenLayoutContainer__GoldenLayoutContainer-dUxNlL jEHdcK",
    },
    "ref": null,
    "rendered": null,
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "className": "GoldenLayoutContainer__GoldenLayoutContainer-dUxNlL jEHdcK",
      },
      "ref": null,
      "rendered": null,
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
